modules/heat_conduction/src/materials/SemiconductorLinearConductivity.C

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SemiconductorLinearConductivity.h"
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<SemiconductorLinearConductivity>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredCoupledVar("temp", "variable for temperature");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.addRequiredParam<Real>("sh_coeff_A", "Steinhart_Hart coefficient A of the material");
+  params.addRequiredParam<Real>("sh_coeff_B", "Steinhart_Hart coefficient B of the material");
```

Comment:
Since you're dividing by this, I think this should be range checked. Use `addRequiredRangeCheckedParam`, and force it to be greater than zero.

Reply:
When I used addRequiredRangeCheckedParam<Real>, I had "invalid float syntax for parameter: Materials/sigma/temp=T" error from test.  Should I use coupledVar?

Reply:
Do this here to ensure that you don't divide by zero for sh_coeff_B:
```
params.addRequiredParam<Real>("sh_coeff_B", "sh_coeff_B > 0", "Steinhart_Hart coefficient B of the material");
```

Reply:
temp is where I have problem with.  It is coupledVar but not able to find addRequiredRangeCheckedCoupledVar.

Reply:
Yeah, there is no such thing.  You can't guarantee that the field variable will be within a certain range, just parameters.